libs/dh/market-participant/actors/feature-organizations/src/lib/drawer/tabs/dh-organization-history.component.ts

```diff
@@ -0,0 +1,116 @@
+ * limitations under the License.
+ */
+import { Component, computed, effect, input } from '@angular/core';
+
+import { TranslocoDirective } from '@ngneat/transloco';
+
+import { WATT_CARD } from '@energinet-datahub/watt/card';
+import { WattDatePipe } from '@energinet-datahub/watt/date';
+import { VaterStackComponent } from '@energinet-datahub/watt/vater';
+import { WattSpinnerComponent } from '@energinet-datahub/watt/spinner';
+import { WattEmptyStateComponent } from '@energinet-datahub/watt/empty-state';
+import { WATT_TABLE, WattTableColumnDef, WattTableDataSource } from '@energinet-datahub/watt/table';
+
+import { lazyQuery } from '@energinet-datahub/dh/shared/util-apollo';
+
+import {
+  GetAuditLogByOrganizationIdDocument,
+  OrganizationAuditedChangeAuditLogDto,
+} from '@energinet-datahub/dh/shared/domain/graphql';
+
+@Component({
+  selector: 'dh-organization-history',
+  standalone: true,
+  template: ` <ng-container
+    *transloco="let t; read: 'marketParticipant.organizationsOverview.drawer'"
+  >
+    <vater-stack [fill]="'horizontal'" [align]="'center'">
+      @if (isLoading()) {
+        <watt-spinner />
+      }
+    </vater-stack>
+    @let auditLogDataSource = auditLog();
+    @if (!isLoading() && auditLogDataSource.data.length === 0) {
+      <watt-empty-state
+        [icon]="hasError() ? 'custom-power' : 'cancel'"
+        size="small"
+        [title]="hasError() ? t('tabs.shared.error') : t('tabs.shared.noData')"
+      />
+    }
+    @if (auditLogDataSource.data.length > 0 && !isLoading()) {
+      <watt-card variant="solid">
+        <watt-table
+          [dataSource]="auditLogDataSource"
+          [columns]="auditLogColumns"
+          [hideColumnHeaders]="true"
+          [suppressRowHoverHighlight]="true"
+          sortBy="timestamp"
+          sortDirection="desc"
+        >
+          <ng-container *wattTableCell="auditLogColumns['timestamp']; let element">
+            {{ element.timestamp | wattDate: 'long' }}
+          </ng-container>
+          <ng-container *wattTableCell="auditLogColumns['value']; let entry">
+            <div [innerHTML]="t('tabs.history.changeTypes.' + entry.change, entry)"></div>
+          </ng-container>
+        </watt-table>
+      </watt-card>
+    }
+  </ng-container>`,
+  imports: [
+    TranslocoDirective,
+
+    VaterStackComponent,
+
+    WATT_TABLE,
+    WATT_CARD,
+    WattDatePipe,
+    WattSpinnerComponent,
+    WattEmptyStateComponent,
+  ],
+})
+export class DhOrganizationHistoryComponent {
+  private getAuditLogByOrganizationIdQuery = lazyQuery(GetAuditLogByOrganizationIdDocument);
+
+  organizationId = input<string>();
+
+  isLoading = this.getAuditLogByOrganizationIdQuery.loading;
+  hasError = computed(() => this.getAuditLogByOrganizationIdQuery.error !== undefined);
+
+  auditLog = computed(
+    () =>
+      new WattTableDataSource<OrganizationAuditedChangeAuditLogDto>(
+        this.getAuditLogByOrganizationIdQuery.data()?.organizationAuditLogs
+      )
+  );
```

Comment:
suggestion: this will create a new `WattTableDataSource` on new data. Material's advice is to update the `.data` property instead. Let's use that. Also let's have an empty array as fallback in case `.data()?.organizationAuditLogs` is `undefined` (as it looks like it could be).

Reply:
Fixed in https://github.com/Energinet-DataHub/greenforce-frontend/pull/3444/commits/32483794bfde07552fe022da0a2129e6dda3deb6